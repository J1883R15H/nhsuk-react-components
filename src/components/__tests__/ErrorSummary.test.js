import React from 'react';
import { shallow, mount } from 'enzyme';
import sinon from 'sinon';
import ErrorSummary from '../error-summary';

describe('ErrorSummary', () => {
  it('item matches snapshot', () => {
    const shallowMount = shallow(
      <ErrorSummary.ErrorItem href="#">Error item</ErrorSummary.ErrorItem>
    );
    expect(shallowMount).toMatchSnapshot();
    shallowMount.unmount();
  });
  it('handles onClick properly', () => {
    const onClick = sinon.spy();
    const shallowMount = shallow(
      <ErrorSummary.ErrorItem onClick={onClick}>
        Error item
      </ErrorSummary.ErrorItem>
    );
    shallowMount.find('a').simulate('click');
    sinon.assert.called(onClick);
    shallowMount.unmount();
  });
  it('all matches snapshot', () => {
    const mountedElement = mount(
      <ErrorSummary
        title="This is a problem"
        description="Optional description of the errors and how to correct them"
      >
        <ErrorSummary.ErrorItem href="#example-error-1">
          Link to error with explanation
        </ErrorSummary.ErrorItem>
        <ErrorSummary.ErrorItem href="#example-error-2">
          Link to error with explanation
        </ErrorSummary.ErrorItem>
      </ErrorSummary>
    );
    expect(mountedElement).toMatchSnapshot();
    mountedElement.unmount();
  });
  it("doesn't render description or title when not provided", () => {
    const mountedElement = mount(
      <ErrorSummary>
        <ErrorSummary.ErrorItem href="#example-error-1">
          Link to error with explanation
        </ErrorSummary.ErrorItem>
        <ErrorSummary.ErrorItem href="#example-error-2">
          Link to error with explanation
        </ErrorSummary.ErrorItem>
      </ErrorSummary>
    );
    expect(mountedElement).toMatchSnapshot();
    mountedElement.unmount();
  });
});
