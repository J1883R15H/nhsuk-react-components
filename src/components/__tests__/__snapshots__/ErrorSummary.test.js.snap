// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorSummary all matches snapshot 1`] = `
<ErrorSummary
  ariaLabelledBy="error-summary-title"
  className=""
  dataModule="error-summary"
  description="Optional description of the errors and how to correct them"
  role="alert"
  style={Object {}}
  title="This is a problem"
>
  <div
    aria-labelledby="error-summary-title"
    className="nhsuk-error-summary"
    data-module="error-summary"
    role="alert"
    style={Object {}}
    tabIndex="-1"
  >
    <h2
      className="nhsuk-error-summary__title"
    >
      This is a problem
    </h2>
    <div
      className="nhsuk-error-summary__body"
    >
      <p>
        Optional description of the errors and how to correct them
      </p>
      <ul
        className="nhsuk-list nhsuk-error-summary__list"
      >
        <ErrorItem
          className=""
          href="#example-error-1"
          onClick={[Function]}
          style={Object {}}
        >
          <li
            className=""
            style={Object {}}
          >
            <a
              href="#example-error-1"
              onClick={[Function]}
            >
              Link to error with explanation
            </a>
          </li>
        </ErrorItem>
        <ErrorItem
          className=""
          href="#example-error-2"
          onClick={[Function]}
          style={Object {}}
        >
          <li
            className=""
            style={Object {}}
          >
            <a
              href="#example-error-2"
              onClick={[Function]}
            >
              Link to error with explanation
            </a>
          </li>
        </ErrorItem>
      </ul>
    </div>
  </div>
</ErrorSummary>
`;

exports[`ErrorSummary doesn't render description or title when not provided 1`] = `
<ErrorSummary
  ariaLabelledBy="error-summary-title"
  className=""
  dataModule="error-summary"
  description=""
  role="alert"
  style={Object {}}
>
  <div
    aria-labelledby="error-summary-title"
    className="nhsuk-error-summary"
    data-module="error-summary"
    role="alert"
    style={Object {}}
    tabIndex="-1"
  >
    <h2
      className="nhsuk-error-summary__title"
    />
    <div
      className="nhsuk-error-summary__body"
    >
      <ul
        className="nhsuk-list nhsuk-error-summary__list"
      >
        <ErrorItem
          className=""
          href="#example-error-1"
          onClick={[Function]}
          style={Object {}}
        >
          <li
            className=""
            style={Object {}}
          >
            <a
              href="#example-error-1"
              onClick={[Function]}
            >
              Link to error with explanation
            </a>
          </li>
        </ErrorItem>
        <ErrorItem
          className=""
          href="#example-error-2"
          onClick={[Function]}
          style={Object {}}
        >
          <li
            className=""
            style={Object {}}
          >
            <a
              href="#example-error-2"
              onClick={[Function]}
            >
              Link to error with explanation
            </a>
          </li>
        </ErrorItem>
      </ul>
    </div>
  </div>
</ErrorSummary>
`;

exports[`ErrorSummary item matches snapshot 1`] = `
<li
  className=""
  style={Object {}}
>
  <a
    href="#"
    onClick={[Function]}
  >
    Error item
  </a>
</li>
`;
